### ISSUE TITLE
some output frames  show colorful distortions, especially in the subtitle area.

### ISSUE THREAD

**Issue by hahahaprince** [xinntao/EDVR#22]
![image](https://user-images.githubusercontent.com/22274077/59021040-b0cc9d80-887d-11e9-8082-3cb50fe8a6cf.png)

**Comment by xinntao** [comment]
Oh, it is a severe artifact. Do these artifacts appear when testing with Vid4 or REDS dataset? 

It may raise from the different distributions of the datasets. But they are usually related to BN layers. However, there is no BN layer in the EDVR model.

**Comment by hahahaprince** [comment]
> Oh, it is a severe artifact. Do these artifacts appear when testing with Vid4 or REDS dataset?
> 
> It may raise from the different distributions of the datasets. But they are usually related to BN layers. However, there is no BN layer in the EDVR model.

Artifacts doesn't appear in Vid4 or REDS dataset, and it mostly appears in unnatural scenes.

**Comment by xinntao** [comment]
It is an interesting problem. It is probably raised from the different distributions of the training data and testing data.

**Comment by hahahaprince** [comment]
> It is an interesting problem. It is probably raised from the different distributions of the training data and testing data.

Hi, xinntao. I found that this artifacts become  more severe when the offset mean is larger than 100. Can you give me some advice about how to guarantee the value of the offset when training? I found increasing input image size (deblur task) can alleviate this situation to some extent.

**Comment by xinntao** [comment]
If the offset mean is large than 100, the offsets must be meaningless. This is an issue of DCN for alignment - unstable training. Usually, we can finetune a pre-trained model with a smaller learning rate for those conv layers that predicts the offsets. 

Sometimes, the too large offset mean occurs occasionally. In this case, we just stop it and resume it from a model whose offset prediction is normal.

**Comment by hahahaprince** [comment]
@xinntao I solved this problem by finetuning your pretrained model using 'train_EDVR_M.yml'. But what surprised me was that I only used one video during the finetune process, monitoring the artifacts slowly disappearing. Then the new model work well on 99% video in my dataset without artifacts( too large offset mean  is basically not present), although these video content vary greatly. So I have two questions:
1. Do you think this is because the model learned how to understand unnatural scene in the the finetune process with ? （I found out that noise is mostly in unnatural areas.）
2. The TSA module is behind the DCN module, so finutuning TSA influence DCN by back propagation? I can't understand why is the TSA module so powerful?

**Comment by hahahaprince** [comment]
@xinntao I solved this problem by finetuning your pretrained model using 'train_EDVR_M.yml'. But what surprised me was that I only used one video during the finetune process, monitoring the artifacts slowly disappearing. Then the new model work well on 99% video in my dataset without artifacts( too large offset mean is basically not present), although these video content vary greatly. So I have two questions:

Do you think this is because the model learned how to understand unnatural scene in the the finetune process with only one video? （I found out that noise is mostly in unnatural areas.）
The TSA module is behind the DCN module, so finutuning TSA influence DCN by back propagation? I can't understand why is the TSA module so powerful?

**Comment by Mukosame** [comment]
+1, same artifacts too.

**Comment by xinntao** [comment]
@hahahaprince 
1) I think so. Before the finetuning process, the pre-trained EDVR network has not seen these data and thus produces artifacts. Although with only one video during finetuning, the model learns to deal with these scenes. 
2) Did you perform the fine-tuning on all the parameters? If you use the `train_EDVR_M.yml`, it will fine tune all the parameters. 

@Mukosame 
You can finetune the pre-trained model using your own data, as @hahahaprince does.

**Comment by sjscotti** [comment]
@hahahaprince I am using the `EDVR_REDS_deblurcomp_L.pth`  model and I am getting similar artifacts in some of the videos I am processing with EDVR.  Are you doing blur_comp processing too, and if so, would you be willing to share your modified model?

**Comment by htleeab** [comment]
I also encounter this artifact without offset > 100.
Do anyone confirm that the artifact produced by DCNv2 or TSA? 
How to visualize Ablation on the PCD or TSA like figure 8 and figure 9 in the paper? Do it need to be computed from 3 layer offset?
https://github.com/xinntao/EDVR/blob/72a0f87a85f7d097dd165ff7736cbe07d1604ea8/codes/models/archs/dcn/deform_conv.py#L282
https://github.com/xinntao/EDVR/blob/72a0f87a85f7d097dd165ff7736cbe07d1604ea8/codes/models/archs/EDVR_arch.py#L200
